starwars_api/models.py

```diff
@@ -55,17 +64,39 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.page = 0 #for page iterator
+        self.objects = [] #an iterable (list) of Film or People to be populated and returned to all
+        self.counter = None
+        self.call_generic = getattr(api_client,'get_{}'.format(self.RESOURCE_NAME))
 
     def __iter__(self):
+        # return self
         pass
-
+        
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
+        #do all the heavy lifting
+        #determine if end of page
+        while self.counter > len(self.objects):
+            #iterate page >> returns 404 error for Stop
+            self.page += 1
+            item = 0 #for item iterator in the objects
+            try: 
+                #iterate through items in pages and append to objects
+                #get results - sample: api_client.get_people(**{'page': 1})
+                page_json_data = self.call_generic(**{'page': self.page})
+                result = page_json_data['results'][item]
```

Comment:
When using api or http in general calls to the server are considered expensive operations. If it is possible you should be caching the results from the entire page in a local object and call the next object from there. Calling the server for every result you need will make the app run slow.